ns1/rest/zones.py

```diff
@@ -21,32 +21,59 @@ class Zones(resource.BaseResource):
         "link",
         "primary_master",
         "tags",
+        "views",
     ]
     BOOL_FIELDS = ["dnssec"]
 
+    ZONEFILE_FIELDS = [
+        "networks",
+        "views",
+    ]
+
     def _buildBody(self, zone, **kwargs):
         body = {}
         body["zone"] = zone
         self._buildStdBody(body, kwargs)
         return body
 
     def import_file(
-        self, zone, zoneFile, callback=None, errback=None, **kwargs
+        self, zone, zoneFile, name=None, callback=None, errback=None, **kwargs
     ):
         files = [("zonefile", (zoneFile, open(zoneFile, "rb"), "text/plain"))]
+        params = self._buildImportParams(kwargs, name=name)
         return self._make_request(
             "PUT",
             "import/zonefile/%s" % (zone),
             files=files,
+            params=params,
             callback=callback,
             errback=errback,
         )
 
-    def create(self, zone, callback=None, errback=None, **kwargs):
+    # Extra import args are specified as query parameters not fields in a JSON object.
+    def _buildImportParams(self, fields, name=None):
+        params = {}
+        # Arrays of values should be passed as multiple instances of the same
+        # parameter but the zonefile API expects parameters containing comma
+        # seperated values.
+        if fields.get("networks") is not None:
+            networksStrs = map(str, fields["networks"])
```

Comment:
I guess this is because people may pass networks as numbers?

Reply:
Our API takes a list of network numbers.
Unfortunately it takes them in a single query parameter of the form `networks=1,2,3` which is *not* how you should pass an array via parameters, as I've found out while working on the other side of our import API, and not how the request library will pass an array. So we need to convert the array to a string.
If allowed to the libs will pass an array using multiple params of the same name:
`networks=1&networks=2&networks=3`
but our API won't accept that.